tests/views/PageSOPortal.spec.js

```diff
@@ -0,0 +1,77 @@
+import { mount } from '@vue/test-utils';
+import { describe, it, expect, vi } from 'vitest';
+import PageSOPortal from '@/views/PageSOPortal.vue';
+
+vi.mock('@/services/ImageUploadService', () => ({
+    default: {
+        uploadImages: vi.fn(() => Promise.resolve({ success: false })),
+    },
+}));
+
+const storeMock = {
+    state: {
+        user: {
+            contactForm: {
+                images: [],
+                cno: '1234567890',
+            },
+        },
+    },
+    dispatch: vi.fn(),
+};
+
+const stubs = {
+    TemplateSOPortal: {
+        template: `
+            <button class="template-submit" @click="$emit('final-submit')">
+                Submit
+            </button>
+        `,
+    },
+    LoaderModal: {
+        template: '<div class="loader-modal"></div>',
+    },
+};
+
+const factory = () =>
+    mount(PageSOPortal, {
+        global: {
+            stubs,
+            mocks: {
+                $store: storeMock,
+                $router: {
+                    push: vi.fn(),
+                },
+                $buefy: {
+                    toast: {
+                        open: vi.fn(),
+                    },
+                },
+            },
+        },
+        data() {
+            return {
+                isLoading: false,
```

Comment:
You mock isLoading: false in data but don't test the LoaderModal:
Add one test for it.